api/src/main/java/org/openmrs/api/handler/PersonSaveHandler.java

```diff
@@ -42,13 +45,16 @@ public void handle(Person person, User creator, Date dateCreated, String other)
 		
 		// address collection
 		if (person.getAddresses() != null && !person.getAddresses().isEmpty()) {
+			Set<Address> blankAddresses = new HashSet<>();
 			for (PersonAddress pAddress : person.getAddresses()) {
-				if (pAddress.isBlank()){
-					person.removeAddress(pAddress);
+				if (pAddress.isBlank()) {
```

Comment:
Is this a bug fix? Or is it a change caused by switching to annotations?

Reply:
Definitely a bug, it's java 101, you can't remove an element from the same Iterable you are looping over, you get a ConcurrentModificationException